src/ucp/core/ucp_listener.c

```diff
@@ -242,9 +242,11 @@ ucp_listen_on_cm(ucp_listener_h listener, const ucp_listener_params_t *params)
     ucs_assert_always(num_cms > 0);
 
     uct_params.field_mask       = UCT_LISTENER_PARAM_FIELD_CONN_REQUEST_CB |
-                                  UCT_LISTENER_PARAM_FIELD_USER_DATA;
+                                  UCT_LISTENER_PARAM_FIELD_USER_DATA |
+                                  UCT_LISTENER_PARAM_FIELD_BACKLOG;
     uct_params.conn_request_cb  = ucp_cm_server_conn_request_cb;
     uct_params.user_data        = listener;
+    uct_params.backlog          = ucs_min(INT_MAX, worker->context->config.ext.listener_backlog);
```

Comment:
maybe use `-1` for `UCS_ULUNITS_AUTO`.
otherwise, it is not obvious, when `UCX_LISTENER_BACKLOG=inf` resulted to the default value for backlog instead of the maximal possible value (i.e. INT_MAX).

Reply:
backlog is an int (from uct's api) so i can't assign an unsigned long to an int..

Reply:
> backlog is an int (from uct's api) so i can't assign an unsigned long to an int..

But it is ULUNITS right now, so user could specify `inf`, this will be `-1` when cast to `int`, right?

Reply:
but i'm not casting to int. if the user sets 'inf', since it's an unsigned long, it will be a very large number and the ucs_min between this number and INT_MAX will be INT_MAX which would be set into the int backlog.

Reply:
ah, I see
I'm okay 👍 
But if this is`auto`, it will be `INT_MAX`, right? I'd expect to make it equal to the system default value in this case

Reply:
right, backlog would be set to INT_MAX here and then, inside uct, will be set to the system's setting.

Reply:
ok, but if this is `inf`, it also would be adjusted to the system default, right?

Reply:
to make sure ucs_min is done right:
`ucs_min((size_t)INT_MAX, worker->context->config.ext.listener_backlog)`;